File: Algorithm/Recursion/779.ts
function kthGrammar(n: number, k: number): number {
  if (n === 1) {
    return 0;
  } else if (k > 2 ** (n - 2)) {
    return 1 - kthGrammar(n - 1, k - 2 ** (n - 2));
  } else {
    return kthGrammar(n - 1, k);
  }
}

File: Algorithm/Recursion/779.py
# K-th Symbol in Grammar
class Solution:
    def kthGrammar(self, n: int, k: int) -> int:
        if n == 1:
            return 0
        elif k > 2 ** (n - 2):
            return 1 - self.kthGrammar(n - 1, k - 2 ** (n - 2))
        else:
            return self.kthGrammar(n - 1, k)
